src/custom-ui/CustomWidgets.js

```diff
@@ -76,4 +77,19 @@ CustomTextWidget.propTypes = {
   schema: PropTypes.object
 }
 
-export const widgets = { checkbox: CustomCheckboxWidget, time: CustomTimeWidget, text: CustomTextWidget }
+/**
+ * Return no SelectWidget when options
+ * are string and number
+ */
+const CustomSelectWidget = (props) => {
+  const options = []
+  for (let i = 0; i < props.options.enumOptions.length; i++) {
+    options.push(props.options.enumOptions[i].label)
+  }
+  if (options.sort().join() !== ['number', 'string'].join()) {
+    const newProps = { ...props, hideLabel: true }
```

Comment:
Could you check that the number input field is the one that is always shown on default by rjsf? I'm concerned that if the string input is shown in some cases, users would not have an option to change it once the select dropdown is hidden.